platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md

```diff
@@ -430,11 +440,56 @@ here, but will be defined within the Sheriff's design documentation.
 
 ### Public API
 
-There are no changes to the public API as a result of this design.
+![Consensus API](consensus-api.png)
 
----
+The public API of the Consensus module forms the interface boundary between Consensus and Execution.
+
+#### initialize
+
+Called by the Execution layer when it creates a new Consensus instance. This call will include any required arguments
+to fully initialized Consensus. For example, it will include any initial state corresponding to the current round
+at which Consensus should begin work (such as the correct set of judges for that particular round). Each module within
+Consensus supports some kind of initialization API that will be called. For example, each module must be initialized
+with the initial roster for the starting round.
+
+#### destroy
+
+Called by the Execution layer when it destroys an existing Consensus instance. This is particularly needed to shut down
+network connections held by Gossip, but could be used to stop executors, background tasks, etc. as necessary.
+
+#### changeRoster
+
+Called by the Execution layer when a new roster is prepared and ready for use. 
+
+#### onBehind
+
+Called by Consensus to notify Execution that the Consensus system is very far behind in processing relative to its
+peers (most likely because it cannot find a peer that contains any of the events needed for advancing consensus).
+Execution will use this call to initiate a reconnect procedure.
+
+#### onBadNode
+
+Called by Consensus to notify Execution of information about bad nodes. This can include both when a node goes "bad",
+or when it is back in the good graces of the Sheriff.
+
+#### onNewEvent
+
+Called by the Event Intake module of Consensus to give the Execution layer a change to fill the event with transactions.
+Note that in the current system the control in inverted -- transactions are submitted by Execution to Consensus, which
+buffers them and includes them in events. As per this design, Consensus will reach out to Execution to ask it for all
+transactions that should be included in the next event, allowing the Execution layer (the application) to decide which
+transactions to include.
+
+#### onStaleEvent
+
+Called by Consensus to notify Execution when an event has become stale, giving Execution a chance to inspect the
+transactions of that stale event, and resubmit those transactions in the next `onNewEvent` if they are still valid.
+
+#### onPreHandleEvent
+
+Called by Consensus once for each event emitted in topological order from Event Intake, giving Execution a chance to
+perform some work before the event even comes to consensus.
 
-## Implementation and Delivery Plan
+#### onRound
 
-How should the proposal be implemented? Is there a necessary order to implementation? What are the stages or phases
-needed for the delivery of capabilities? What configuration flags will be used to manage deployment of capability? 
+Called by Consensus once for each round that comes to consensus.
```

Comment:
There's nothing necessarily wrong with this, but note that the current system outputs a list of rounds, since it is possible for multiple rounds to come to consensus based on a single event input to the consensus engine.